core/src/main/scala/org/ensime/core/Refactoring.scala

```diff
@@ -184,8 +187,20 @@ trait RefactoringImpl {
           reloadAndType(file)
           doInlineLocal(procId, tpe, file, start, end)
         case RenameRefactorDesc(newName, file, start, end) =>
-          reloadAndType(file)
-          doRename(procId, tpe, newName, file, start, end)
+          import scala.reflect.internal.util.{ RangePosition, OffsetPosition }
+          val sourceFile = createSourceFile(file.getPath)
+          askLoadedTyped(sourceFile)
```

Comment:
we seem to be doing rather a lot of blocking in this actor... I suppose that's ok because we can't use the PC from other responses anyway. But we definitely shouldn't be blocking on the DB, since this actor manages PC state.